src/cprt_planner_plugins/src/event_horizon_planner.cpp

```diff
@@ -0,0 +1,206 @@
+#include "cprt_planner_plugins/event_horizon_planner.hpp"
+
+#include <cmath>
+#include <memory>
+#include <string>
+
+#include "nav2_util/node_utils.hpp"
+
+namespace cprt_planner_plugins {
+
+EventHorizonPlanner::EventHorizonPlanner()
+    : lp_loader_("nav2_core", "nav2_core::GlobalPlanner"),
+      primary_planner_(nullptr) {}
+
+void EventHorizonPlanner::configure(
+    const rclcpp_lifecycle::LifecycleNode::WeakPtr& parent, std::string name,
+    std::shared_ptr<tf2_ros::Buffer> tf,
+    std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros) {
+  node_ = parent.lock();
+  name_ = name;
+  tf_ = tf;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+  logger_ = node_->get_logger();
+
+  std::string primary_planner;
+
+  // Parameter initialization
+
+  // Interpolation resolution
+  nav2_util::declare_parameter_if_not_declared(
+      node_, name_ + ".interpolation_resolution",
+      rclcpp::ParameterValue(DEFAULT_INTERPOLATION_RESOLUTION));
+  node_->get_parameter(name_ + ".interpolation_resolution",
+                       interpolation_resolution_);
+
+  // Primary Planner
+  nav2_util::declare_parameter_if_not_declared(
+      node_, name_ + ".primary_planner", rclcpp::PARAMETER_STRING);
+  primary_planner =
+      node_->get_parameter(name_ + ".primary_planner").as_string();
+
+  // Horizon Distance
+  nav2_util::declare_parameter_if_not_declared(
+      node_, name_ + ".horizon_distance",
+      rclcpp::ParameterValue(DEFAULT_HORIZON_DISTANCE));
+  node_->get_parameter(name_ + ".horizon_distance", horizon_distance_);
+
+  try {
+    primary_planner_ = lp_loader_.createUniqueInstance(primary_planner);
+  } catch (const pluginlib::PluginlibException& ex) {
+    RCLCPP_FATAL(logger_, "Failed to create planner instance. Exception: %s",
+                 ex.what());
+    return;
+  }
+
+  primary_planner_->configure(parent, name, tf, costmap_ros);
+
+  new_goal_publisher_ =
+      node_->create_publisher<geometry_msgs::msg::PoseStamped>(
+          "intermediate_goal", 10);
+}
+
+// taken from ros-navigation/navigation2_tutorials
+void EventHorizonPlanner::cleanup() {
+  RCLCPP_INFO(logger_, "CleaningUp plugin %s of type EventHorizonPlanner",
+              name_.c_str());
+  primary_planner_->cleanup();
+}
+
+// taken from ros-navigation/navigation2_tutorials
+void EventHorizonPlanner::activate() {
+  RCLCPP_INFO(logger_, "Activating plugin %s of type EventHorizonPlanner",
+              name_.c_str());
+  if (primary_planner_) {
+    RCLCPP_INFO(logger_, "Activate successful");
+  } else {
+    RCLCPP_INFO(logger_, "Planner null");
+  }
```

Comment:
```suggestion
  if (!primary_planner_) {
    RCLCPP_FATAL(logger_, "primary_planner_ is null in activate(). Aborting.");
    std::terminate();
  }  
```
  primary_planner_ being null here is fatal. It is better to crash in a controlled manner than allow a segfault to happen